android/expoview/src/main/java/host/exp/exponent/ExpoUpdatesAppLoader.java

```diff
@@ -53,18 +53,36 @@
   private String mManifestUrl;
   private final boolean mUseCacheOnly;
 
-  private Context mContext;
+  private UpdatesConfiguration mUpdatesConfiguration;
+  private File mUpdatesDirectory;
+  private SelectionPolicy mSelectionPolicy;
+  private Launcher mLauncher;
 
   public ExpoUpdatesAppLoader(String manifestUrl) {
-    this(manifestUrl, false, null);
+    this(manifestUrl, false);
   }
 
-  public ExpoUpdatesAppLoader(String manifestUrl, boolean useCacheOnly, Context context) {
+  public ExpoUpdatesAppLoader(String manifestUrl, boolean useCacheOnly) {
     NativeModuleDepsProvider.getInstance().inject(ExpoUpdatesAppLoader.class, this);
 
     mManifestUrl = manifestUrl;
     mUseCacheOnly = useCacheOnly;
-    mContext = context;
+  }
+
+  public UpdatesConfiguration getUpdatesConfiguration() {
+    return mUpdatesConfiguration;
+  }
+
+  public File getUpdatesDirectory() {
+    return mUpdatesDirectory;
+  }
+
+  public SelectionPolicy getSelectionPolicy() {
+    return mSelectionPolicy;
+  }
+
+  public Launcher getLauncher() {
+    return mLauncher;
   }
```

Comment:
Might not hurt to have these throw if they are null (maybe IllegalStateException) so that we get a failure earlier than an NPE later.